.travis.yml

```diff
@@ -76,32 +78,37 @@ matrix:
             - *common_deps
             - g++-6
 
+env:
+  global:
+    - BUILD_TYPE=Debug CPP_VERSION=14 ASAN=On
```

Comment:
I wonder if we really need the CPP_VERSION variable.  We make too much use of C++14 features to compile with `-std=c++11 -pedantic -Werror`.